presto-main/src/main/java/io/prestosql/eventlistener/EventListenerManager.java

```diff
@@ -39,75 +41,97 @@
 public class EventListenerManager
 {
     private static final Logger log = Logger.get(EventListenerManager.class);
-
     private static final File CONFIG_FILE = new File("etc/event-listener.properties");
-    private static final String NAME_PROPERTY = "event-listener.name";
-
+    private static final String EVENT_LISTENER_NAME_PROPERTY = "event-listener.name";
+    private final List<File> configFiles;
     private final Map<String, EventListenerFactory> eventListenerFactories = new ConcurrentHashMap<>();
-    private final AtomicReference<Optional<EventListener>> configuredEventListener = new AtomicReference<>(Optional.empty());
+    private final AtomicReference<List<EventListener>> configuredEventListeners =
+            new AtomicReference<>(ImmutableList.of());
+
+    @Inject
+    public EventListenerManager(EventListenerConfig config)
+    {
+        this.configFiles = ImmutableList.copyOf(config.getEventListenerFiles());
+    }
 
     public void addEventListenerFactory(EventListenerFactory eventListenerFactory)
     {
         requireNonNull(eventListenerFactory, "eventListenerFactory is null");
 
         if (eventListenerFactories.putIfAbsent(eventListenerFactory.getName(), eventListenerFactory) != null) {
-            throw new IllegalArgumentException(format("Event listener '%s' is already registered", eventListenerFactory.getName()));
+            throw new IllegalArgumentException(
+                format("Event listener '%s' is already registered", eventListenerFactory.getName()));
         }
     }
 
-    public void loadConfiguredEventListener()
-            throws Exception
+    public void loadConfiguredEventListeners()
     {
-        File configFile = CONFIG_FILE.getAbsoluteFile();
-        if (!configFile.exists()) {
-            return;
+        List<File> configFiles = this.configFiles;
+        if (configFiles.isEmpty()) {
+            if (!CONFIG_FILE.exists()) {
+                return;
+            }
+            configFiles = ImmutableList.of(CONFIG_FILE);
         }
-
```

Comment:
Just to be safe, let's add an `AtomicBoolean` so that we can't accidentally load twice
```java
checkState(loading.compareAndSet(false, true), "Event listeners already loaded");